### PR TITLE
[SDK 2665] Update endpoint methods to support 'from' and 'take' checkpoint pagination parameters, where appropriate

### PR DESCRIPTION
**Pull request by evansims**
### Changes

This PR updates the following methods to support checkpoint pagination API2 parameters:

- `Roles::list_users()`
- `Organizations::all_organizations()`
- `Organizations::all_organization_members()`

### References

- Please see the internal Jira tickets SDK-2666 and related epic SDK-2653 for details on this initiative.

### Testing

- [x] This change adds unit test coverage
- [ ] This change adds integration test coverage
- [x] This change has been tested on the latest version of the platform/language or why not

### Checklist

- [x] I have read the [Auth0 general contribution guidelines](https://github.com/auth0/open-source-template/blob/master/GENERAL-CONTRIBUTING.md)
- [x] I have read the [Auth0 Code of Conduct](https://github.com/auth0/open-source-template/blob/master/CODE-OF-CONDUCT.md)
- [x] All existing and new tests complete without errors

### CODE REVIEW

File: auth0/v3/management/organizations.py [hunk 1]
```diff
@@ -31,21 +31,35 @@ def _url(self, *args):
...
 
-           per_page (int, optional): The amount of entries per page. When not set,
-              the default value is up to the server.
+            per_page (int, optional): The amount of entries per page. When not set,
+                the default value is up to the server.
+
+            include_totals (bool, optional): True if the query summary is
+                to be included in the result, False otherwise. Defaults to True.
+
+            from_param (str, optional): Checkpoint Id form which to begin retrieving results.
```
Comment (lbalmaceda) [inline auth0/v3/management/organizations.py:47]:
```suggestion
            from_param (str, optional): Checkpoint Id from which to begin retrieving results.
```

File: auth0/v3/management/organizations.py [hunk 2]
```diff
@@ -31,21 +31,35 @@ def _url(self, *args):
...
+                You can limit the amount of logs using the take parameter.
```
Comment (lbalmaceda) [inline auth0/v3/management/organizations.py:48]: Thought of using "organizations" but it's easier to copy/paste this again and keep making that mistake.
```suggestion
                You can limit the number of entries using the take parameter.
```

File: auth0/v3/management/organizations.py [hunk 3]
```diff
@@ -31,21 +31,35 @@ def _url(self, *args):
         return url
 
     # Organizations
-    def all_organizations(self, page=None, per_page=None):
+    def all_organizations(self, page=None, per_page=None, include_totals=True, from_param=None, take=None):
```
Comment (lbalmaceda) [inline auth0/v3/management/organizations.py:34]: Is `from` a reserved keyword? Note, if changed, the docs would need to be updated.
```suggestion
    def all_organizations(self, page=None, per_page=None, include_totals=True, from=None, take=None):
```

File: auth0/v3/management/organizations.py [hunk 4]
```diff
@@ -186,23 +200,37 @@ def delete_organization_connection(self, id, connection_id):
...
 
-           per_page (int, optional): The amount of entries per page. When not set,
-              the default value is up to the server.
+            per_page (int, optional): The amount of entries per page. When not set,
+                the default value is up to the server.
+
+            include_totals (bool, optional): True if the query summary is
+                to be included in the result, False otherwise. Defaults to True.
+
+            from_param (str, optional): Checkpoint Id form which to begin retrieving results.
```
Comment (lbalmaceda) [inline auth0/v3/management/organizations.py:218]:
```suggestion
            from_param (str, optional): Checkpoint Id from which to begin retrieving results.
```

File: auth0/v3/management/organizations.py [hunk 5]
```diff
@@ -186,23 +200,37 @@ def delete_organization_connection(self, id, connection_id):
...
+                You can limit the amount of logs using the take parameter.
```
Comment (lbalmaceda) [inline auth0/v3/management/organizations.py:219]:
```suggestion
                You can limit the number of entries using the take parameter.
```

File: auth0/v3/management/roles.py [hunk 6]
```diff
@@ -113,14 +113,23 @@ def list_users(self, id, page=0, per_page=25, include_totals=True):
             include_totals (bool, optional): True if the query summary is
                 to be included in the result, False otherwise. Defaults to True.
 
+            from_param (str, optional): Checkpoint Id form which to begin retrieving results.
```
Comment (lbalmaceda) [inline auth0/v3/management/roles.py:116]:
```suggestion
            from_param (str, optional): Checkpoint Id from which to begin retrieving results.
```

File: auth0/v3/management/roles.py [hunk 7]
```diff
@@ -113,14 +113,23 @@ def list_users(self, id, page=0, per_page=25, include_totals=True):
...
+                You can limit the amount of logs using the take parameter.
```
Comment (lbalmaceda) [inline auth0/v3/management/roles.py:117]:
```suggestion
                You can limit the number of entries using the take parameter.
```

**Review by lbalmaceda — CHANGES_REQUESTED** [review]
I'm seeing now that `page/per_page` could mean the same as `from/take`. Have you tried what happens if a call that was using the former now uses the latter? Because if that's the case, all these methods could be simplified in either of these two ways:
- duplicated method signatures, each using its variant.
- keep the old signature, but when building the request, mapping the old variant to the new one.

Might be worth checking with the team if the old properties are deprecated now that the endpoint supports "checkpoint" pagination.

File: auth0/v3/management/organizations.py (hunk 3 above)
Comment (evansims) [reply auth0/v3/management/organizations.py:34]: Correct, `from` is a reserved keyword in Python 3

**Comment by evansims** [comment]
Hey @lbalmaceda 👋  Thanks, I've merged those suggestions.

> I'm seeing now that page/per_page could mean the same as from/take.

It's true `per_page` and `take` mean effectively the same thing, and each is separately required to activate their respective pagination styles. `page` and `from` are unique: `page` is just an incrementing int for the next page of results, whereas `from` is (usually) an API-provided UUID to retrieve the next list of results (although /logs treats this slightly differently.) You make the initial request without a `from`, retrieve the value of `next` returned from the API response, and make your subsequent request with that value as the `from`.

This isn't something that could be mapped in the way suggested, unfortunately.

As far as splitting up the endpoint methods for checkpoint pagination variants, we can certainly do that. Personally, this feels like something that should just be handled with method arguments to me — but I'm happy to implement this however you think best suits the SDK.

> Have you tried what happens if a call that was using the former now uses the latter?

Checkpoint pagination always takes precedence by the API in handling requests; it falls back to basic pagination when `take` is missing.

> Might be worth checking with the team if the old properties are deprecated now that the endpoint supports "checkpoint" pagination.

No indication from anyone I've spoken with that basic pagination is going anywhere. Checkpoint pagination just addresses the problem of API dataset changes being applied between client API requests, causing a shift in results, which can lead to gaps in data on the client's end.

**Review by lbalmaceda — APPROVED** [review]

**Pull request merged**